### PR TITLE
Fixed the isChild and isVisibileForParent properties.

### LINKED ISSUE #89: profileInfo.isChild is always false
profileInfo.isChild is set to false, even if it is a child profile. This leads to a crash when doing m.children() while logged in as a pupil.

### PR DESCRIPTION
**Pull request by Argetan**
I changed the way we check if someone is a child and if their parents can view their info according to what was discussed in #120 and #89, it should close #89.

### CODE REVIEW

File: src/magister.js [hunk 1]
```diff
@@ -426,8 +426,21 @@ class Magister {
 			self.profileInfo = new ProfileInfo(self, accountData.Persoon)
 			self._privileges = new Privileges(self, accountData.Groep[0].Privileges)
 
+			// REVIEW: Maybe pass this as an argument to the ProfileInfo constructor?
+			// If someone has this privilege then they are a parent
```
Comment (lieuwex) [inline src/magister.js]: Let's do this! A nice third parameter would be `isParent` or something.

File: src/magister.js [hunk 2]
```diff
@@ -426,8 +426,21 @@ class Magister {
...
+			self.profileInfo.isChild = !(await self._privileges.can("kinderen", "read"))
+
 			self._personUrl = `${schoolUrl}/api/personen/${id}`
 			self._pupilUrl = `${schoolUrl}/api/leerlingen/${id}`
+
+			// REVIEW: Maybe pass this as an argument to the ProfileInfo constructor?
+			// If we are a child then check if our parents can see our information
+			if (self.profileInfo.isChild) {
+				self.profileInfo.isVisibleForParent =
```
Comment (lieuwex) [inline src/magister.js]: I think we can just remove this property from the `ProfileInfo` class completely, and just make it a method which does the `autorisatie` check.

Comment (lieuwex) [reply src/magister.js]: (an async function which returns a `bool`, and always false if `!isParent`)

**Comment by 7kasper** [comment]
I vouch for isParent too, as it seems you can also login with a teacher account 😉

**Comment by Argetan** [comment]
I was just about to implement the suggestions, however, all of a sudden tests keep failing because of an `AuthError: Invalid username`, anyone else having that issue? It was fine before I went to get dinner.

**Comment by Sj3rd** [comment]
Please check the authCode your using.

**Comment by Argetan** [comment]
My bad, completely forgot about that, thanks!

**Comment by Argetan** [comment]
Perhaps we should check if a teacher also has the `kinderen read` privilege, because if so we need to figure out a way to distinguish teachers form children probably.

**Comment by lieuwex** [comment]
Thanks! merged it as cb4094f since the styling was a bit incorrect. Please run `npm run lint`, or even `npm run lint --fix` (which will fix it automatically). Which will clean up to code to be consistent with the rest of the codebase, but it's not a problem for me to go over it manually. :)

**Pull request closed without merging**